### PR TITLE
chore: prepare release v9

### LINKED ISSUE #951: Compute transaction RTT again

### LINKED ISSUE #1518: Measure command time, time out responses accordingly
During a node capability discovery or interview, as well as traffic generated due to user activation, a
controlling nodes MUST timeout waiting for responses (reports) as part of the capability discovery or
controlling scenarios.
Two timers named CommandTime and ReportTime are used for timing out during a node discovery
interview. Illustrations are given for secure and non-secure cases in Figure 1 and Figure 2
* CommandTime is measured by the application
* ReportTime timeout SHOULD be set to CommandTime + 1 second.

grafik

### LINKED ISSUE #3091: Merge `.json` cache file into `.values.jsonl`
Now that the JSONL DB is actually robust, the monolithic `.json` file is the weak link. We should merge both.

### LINKED ISSUE #3120: Investigate migration from lerna again
https://github.com/Dcard/yarn-plugins seems to be fixed to work with Yarn 3 now.

### LINKED ISSUE #3651: Rename controller methods and properties in the next major version
1. serialApiGt(e), serialApiLt(e) --> sdkVersionGt(e), sdkVersionLt(e) - because this references the Z-Wave SDK version, not the serial API Version
2. serialApiVersion -> firmwareVersion - this references the controller's firmware version, not the serial API version
3. libraryVersion -> sdkVersion - for consistency

### LINKED ISSUE #4272: zwave-js override global Promise rejection mechanism which cause side-effects
### Is your problem within Home Assistant (Core or Z-Wave JS Integration)?

NO, my problem is NOT within Home Assistant or the ZWave JS integration

### Is your problem within ZWaveJS2MQTT?

NO, my problem is NOT within ZWaveJS2MQTT

### Checklist

- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there.

### Describe the bug

Hi,

I'm the core maintainer of the open-source project [Gladys Assistant](https://github.com/GladysAssistant/Gladys), and we are trying to migrate to this library for our Z-Wave integration.

First, congrats on the work on this great lib 🙂

**What causes the bug?**

It seems that this library loads a third-party NPM module, [@sentry/utils](https://www.npmjs.com/package/@sentry/utils), a library that is not intended for production use and that override some Promise rejection behavior in the global scope.

It causes `unHandledPromiseRejection` to be handled differently as soon as we require the `zwave-js` module, and cause side-effects with other librairies.

**Steps to reproduce the behavior**

I managed to write a very simple script to reproduce the issue: 

```javascript
async function start() {
  throw new Error("error");
}

(async () => {
  // require("zwave-js");
  start();
})();
```

If you execute this script with the require commented, you'll get this result (in Node 14): 

```
(node:14305) UnhandledPromiseRejectionWarning: Error: error
    at start (/Users/pierregilles/sandbox/zwavejs/index.js:2:9)
    at /Users/pierregilles/sandbox/zwavejs/index.js:7:3
    at Object.<anonymous> (/Users/pierregilles/sandbox/zwavejs/index.js:8:3)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:75:12)
    at internal/main/run_main_module.js:17:47
(Use `node --trace-warnings ...` to show where the warning was created)
(node:14305) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:14305) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

As soon as I uncomment the require, I get this result: 

``` 
This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
Error: error
    at start (/Users/pierregilles/sandbox/zwavejs/index.js:2:9)
    at /Users/pierregilles/sandbox/zwavejs/index.js:7:3
    at Object.<anonymous> (/Users/pierregilles/sandbox/zwavejs/index.js:8:3)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:75:12)
    at internal/main/run_main_module.js:17:47
```

As you can see, the zwave-js library mutate a global object and cause errors to be thrown differently.

### Which branches or versions?

version:
`node-zwave-js` branch: 8.11.6

### PR DESCRIPTION
**Pull request by AlCalzone**
This PR is meant to collect the changes that will go into the upcoming v9 release.

fixes: #951
fixes: #1518
closes: #3651
closes:  #3120
fixes: #4272
fixes: #3091

### CODE REVIEW

File: docs/getting-started/migrating-to-v9.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Migrating to v9
+
+The recent rewrite of the message queue made it much simpler to support use cases that go beyond executing a simple Serial API command.
+Examples are requesting Security S0 nonces (which was relatively convoluted before), handling other multi-step commands, as well as the ability to time individual commands.
+
+The latter allows us to be more spec-compliant in that we now time out much faster while waiting for a response to a GET request. Since this required a breaking chance, we use the opportunity to collect some more breaking changes in version 9.x.
```
Comment (dfederm) [inline docs/getting-started/migrating-to-v9.md]: Spelling nit: "change"

File: packages/zwave-js/src/lib/driver/MessageGenerators.ts [hunk 2]
```diff
@@ -31,16 +31,20 @@ export type MessageGeneratorImplementation = (
 	 * without waiting for the message generator to finish completely.
 	 */
 	onMessageSent: (msg: Message, result: Message | undefined) => void,
+
+	/** Can be used to extend the timeout wa */
```
Comment (dfederm) [inline packages/zwave-js/src/lib/driver/MessageGenerators.ts]: Nit: "wa"?

**Comment by AlCalzone** [comment]
@zwave-js-bot pack this

**Comment by AlCalzone** [comment]
@marcus-j-davies lets see if this works

**Comment by marcus-j-davies** [comment]
Nice! - will dig into this shortly.

**Comment by marcus-j-davies** [comment]
I guess not 😆.
its fine - no rush 👍

**Comment by AlCalzone** [comment]
Ok I'll try tomorrow after a rebase

**Comment by AlCalzone** [comment]
@zwave-js-bot pack this

**Comment by AlCalzone** [comment]
@zwave-js-bot pack this

**Comment by AlCalzone** [comment]
@zwave-js-bot pack this

**Comment by AlCalzone** [comment]
@zwave-js-bot pack this

**Comment by AlCalzone** [comment]
@zwave-js-bot pack this

**Comment by AlCalzone** [comment]
@zwave-js-bot pack this

**Comment by AlCalzone** [comment]
@marcus-j-davies took long enough, but I figured it out 🎉

**Comment by marcus-j-davies** [comment]
Nice!

I'll start playing with this shortly. Thanks.

**Comment by marcus-j-davies** [comment]
@AlCalzone 

image

Given **lwr** has no repeaters - would one look at that as a direct connection?

this is awesome BTW!
Its going to enable maps to be really tight!

Feel free to mark this as off-topic 😉

**Comment by AlCalzone** [comment]
That's right

**Comment by AlCalzone** [comment]
@zwave-js-bot pack this

**Comment by AlCalzone** [comment]
@zwave-js-bot pack this

**Comment by AlCalzone** [comment]
@zwave-js-bot pack this

**Comment by AlCalzone** [comment]
@zwave-js-bot pack this

**Pull request merged**